### PR TITLE
Improve Webpack build time and Dev mode build time

### LINKED ISSUE #397: Development time dramatically hangs while in dev mode
**Describe the bug**
When in dev mode, the build hangs at optimisation process for too long.

Screenshot 2020-02-10 at 20 22 20

**To Reproduce**
1. Trigger a change event in the app while in dev mode
2. Monitor the console.

**Expected behavior**
Shouldn't hang in dev mode that long.

### PR DESCRIPTION
**Pull request by DimalChandrasiri**
## Purpose
Thi PR will fix #397 and #396 as well as improve the build time of the webpack build by using multiple threads to transpile Typescript. 

## Goals
- Improve Development build time by running the minimizer option only in production mode builds.
- Improve overall build time by adding threads to TS compiler.
- Added `Hot Module Replacement` ability to `admin portal`

### CODE REVIEW

File: apps/admin-portal/src/index.tsx [hunk 1]
```diff
@@ -37,3 +37,8 @@ ReactDOM.render(
     ),
     document.getElementById("root")
 );
+
+//Accept HMR for updated modules
+if (module && module.hot) {
+    module.hot.accept();
+}
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/index.tsx]: Add new line

File: apps/admin-portal/package.json [hunk 2]
```diff
@@ -29,7 +29,9 @@
         "@wso2is/validation": "^1.0.2"
     },
     "devDependencies": {
-        "run-script-os": "^1.0.7"
+        "fork-ts-checker-webpack-plugin": "^4.0.3",
```
Comment (jeradrutnam) [inline apps/admin-portal/package.json]: Shall we pull it from the root package.json? And use as a peerDependency.

File: apps/admin-portal/package.json [hunk 3]
```diff
@@ -29,7 +29,9 @@
...
+        "run-script-os": "^1.0.7",
+        "thread-loader": "^2.1.3"
```
Comment (jeradrutnam) [inline apps/admin-portal/package.json]: This as well. As these will be use commonly in other packages.

**Comment by jeradrutnam** [comment]
Add the changes to `user-portal` as well

File: apps/admin-portal/package.json (hunk 3 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/package.json]: Resolved with commit `Fix peer dependencies` commit.

File: apps/admin-portal/package.json (hunk 2 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/package.json]: Resolved with commit `Fix peer dependencies` commit.

File: apps/admin-portal/src/index.tsx (hunk 1 above)
Comment (DimalChandrasiri) [reply apps/admin-portal/src/index.tsx]: Resolved with commit `Fix peer dependencies` commit.

**Pull request merged**